__tests__/integration.js

```diff
@@ -58,11 +58,13 @@ const PORT_RANGE = MAX_PORT_NUM - MIN_PORT_NUM;
 
 const getRandomPort = () => Math.floor(Math.random() * PORT_RANGE) + MIN_PORT_NUM;
 
-function runYarn(args: Array<string> = [], options: Object = {}): Promise<Array<Buffer>> {
-  const {stderr, stdout} = execa(path.resolve(__dirname, '../bin/yarn'), args, options);
+async function runYarn(args: Array<string> = [], options: Object = {}): Promise<Array<Buffer>> {
+  const process = execa(path.resolve(__dirname, '../bin/yarn'), args, options);
 
-  const stdoutPromise = misc.consumeStream(stdout);
-  const stderrPromise = misc.consumeStream(stderr);
+  const stdoutPromise = misc.consumeStream(process.stdout);
+  const stderrPromise = misc.consumeStream(process.stderr);
+
+  await process;
```

Comment:
This doesn't look right to me since it looks like `stdout` and `stderr` are only available after `await`ing `process`: https://github.com/sindresorhus/execa/blob/master/index.js#L259

I guess the following is what we need:
```
  const {stdout, stderr} = await execa(path.resolve(__dirname, '../bin/yarn'), args, options);
  
  return Promise.all([
    misc.consumeStream(stdout),
    misc.consumeStream(stderr),
  ]);
```

If I'm missing something, may be a comment here explaining the ordering would help?